doc/user-guide/data-structures.rst

```diff
@@ -495,6 +497,314 @@ dimension and non-dimension variables:
     ds.coords["day"] = ("time", [6, 7, 8, 9])
     ds.swap_dims({"time": "day"})
 
+DataTree
+--------
+
+:py:class:`~xarray.DataTree` is ``xarray``'s highest-level data structure, able to
+organise heterogeneous data which could not be stored inside a single
+:py:class:`~xarray.Dataset` object. This includes representing the recursive structure
+of multiple `groups`_ within a netCDF file or `Zarr Store`_.
+
+.. _groups: https://www.unidata.ucar.edu/software/netcdf/workshops/2011/groups-types/GroupsIntro.html
+.. _Zarr Store: https://zarr.readthedocs.io/en/stable/tutorial.html#groups
+
+Each :py:class:`~xarray.DataTree` object (or "node") contains the same data that a single
+:py:class:`xarray.Dataset` would (i.e. :py:class:`~xarray.DataArray` objects stored under hashable
+keys), and so has the same key properties:
+
+- ``dims``: a dictionary mapping of dimension names to lengths, for the
+  variables in this node, and this node's ancestors,
+- ``data_vars``: a dict-like container of DataArrays corresponding to variables
+  in this node,
+- ``coords``: another dict-like container of DataArrays, corresponding to
+  coordinate variables in this node, and this node's ancestors,
+- ``attrs``: dict to hold arbitrary metadata relevant to data in this node.
+
+A single :py:class:`~xarray.DataTree` object acts much like a single :py:class:`~xarray.Dataset` object, and
+has a similar set of dict-like methods defined upon it. However, :py:class:`~xarray.DataTree`\s
+can also contain other :py:class:`~xarray.DataTree` objects, so they can be thought of as
+nested dict-like containers of both :py:class:`xarray.DataArray`\s and :py:class:`~xarray.DataTree`\s.
+
+A single datatree object is known as a "node", and its position relative to
+other nodes is defined by two more key properties:
+
+- ``children``: An ordered dictionary mapping from names to other :py:class:`~xarray.DataTree`
+  objects, known as its "child nodes".
+- ``parent``: The single :py:class:`~xarray.DataTree` object whose children this datatree is a
+  member of, known as its "parent node".
+
+Each child automatically knows about its parent node, and a node without a
+parent is known as a "root" node (represented by the ``parent`` attribute
+pointing to ``None``). Nodes can have multiple children, but as each child node
+has at most one parent, there can only ever be one root node in a given tree.
+
+The overall structure is technically a `connected acyclic undirected rooted graph`,
+otherwise known as a `"Tree" <https://en.wikipedia.org/wiki/Tree_(graph_theory)>`_.
+
+.. note::
+
+    Technically a :py:class:`~xarray.DataTree` with more than one child node forms an
+    `"Ordered Tree" <https://en.wikipedia.org/wiki/Tree_(graph_theory)#Ordered_tree>`_,
+    because the children are stored in an Ordered Dictionary. However, this
+    distinction only really matters for a few edge cases involving operations
+    on multiple trees simultaneously, and can safely be ignored by most users.
```

Comment:
I might drop this, or cut it down into a short inline parenthetical. We don't go into the detailed around ordering for Dataset variables or coordinates, it just isn't something that users really need to think about.

Reply:
dropped.